Fix unexpected IOKit use on Catalyst

pvollan opened the pull request:
#### 1f83e37bcedff3b85f25ce17e58f4487c352bbbf
<pre>
Fix unexpected IOKit use on Catalyst
<a href="https://bugs.webkit.org/show_bug.cgi?id=252247">https://bugs.webkit.org/show_bug.cgi?id=252247</a>
rdar://105453976

Reviewed by Simon Fraser.

If present, use the cached value for PPI provided by the UI process.

* Source/WebCore/platform/ios/PlatformScreenIOS.mm:
(WebCore::screenPPIFactor):

Canonical link: <a href="https://commits.webkit.org/260547@main">https://commits.webkit.org/260547@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/40c54b26ed3e3b6df9d1a6205ba2735f17c4b465

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107735 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16793 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40637 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116885 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116271 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111625 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18212 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8111 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99917 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113489 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13753 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96951 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41436 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95661 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28591 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/83265 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9764 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29939 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10447 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6828 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15918 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49523 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7291 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12009 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/platform/ios/PlatformScreenIOS.mm:
```diff
@@ -129,6 +129,9 @@ FloatRect screenAvailableRect(Widget* widget)
 
 float screenPPIFactor()
 {
+    if (auto data = screenData(primaryScreenDisplayID()))
```
Can we just ensure that we always have screenData instead of having two code paths?

pvollan replied:
> Can we just ensure that we always have screenData instead of having two code paths?

Yes, that is a good point. Currently, the screenPPIFactor function is called from WebCore, which does not have knowledge about which process it is being called from. I wonder if we should send the screen data to all WebKit processes to make sure there always is a cached version? We would also need to cache the screen data in WK1.

Thanks for reviewing!

pvollan committed:
Fix unexpected IOKit use on Catalyst
https://bugs.webkit.org/show_bug.cgi?id=252247
rdar://105453976

Reviewed by Simon Fraser.

If present, use the cached value for PPI provided by the UI process.

* Source/WebCore/platform/ios/PlatformScreenIOS.mm:
(WebCore::screenPPIFactor):

Canonical link: https://commits.webkit.org/260547@main
